api/wasm.go

```diff
@@ -151,9 +151,13 @@ type Module interface {
 
 	// ExportedFunction returns a function exported from this module or nil if it wasn't.
 	//
-	// Note: The default wazero.ModuleConfig attempts to invoke `_start`, which
-	// in rare cases can close the module. When in doubt, check IsClosed prior
-	// to invoking a function export after instantiation.
+	// # Notes
+	//   - The default wazero.ModuleConfig attempts to invoke `_start`, which
+	//     in rare cases can close the module. When in doubt, check IsClosed prior
+	//     to invoking a function export after instantiation.
+	//   - Because of the semantic of host function that it assumes the existence of "importing module" and it has access to
+	//     the memory of the importing module for example, direct use of ExportedFunction is forbidden for host modules.
+	//     Practically speaking, it is usually meaningless to directly call a host function from Go code as it is already somewhere in Go code.
```

Comment:
```suggestion
	//   - Because the semantics of host functions assumes the existence of an "importing module", and because the host function has access to
	//     the memory of the importing module, direct use of ExportedFunction is forbidden for host modules.
	//     Practically speaking, it is usually meaningless to directly call a host function from Go code as it is already somewhere in Go code.
```

nit, I think this slight rephrasing should be clearer. Same nit in builder.go